core/local/atom/initial_diff.js

```diff
@@ -89,6 +89,10 @@ async function initialState(
   const byInode /*: Map<number|string, Metadata> */ = new Map()
   const docs /*: Metadata[] */ = await opts.pouch.byRecursivePathAsync('')
   for (const doc of docs) {
+    if (metadata.needsRevMigration(doc)) {
```

Comment:
I wouldn't have put the migration inside the local watcher. Instead, I'd have added a design doc to pouch with a schema version, and when the cozy-desktop starts and initializes the database, it loads this document and if the schema version is too old, it plays the migration(s) from the old schema version up to the current schema version. And only after that, you can call `sync`.

I'm not saying that my method is better, or that you must change to it. It's just a personal preference to not have the migration stuff inside the watcher logic.

Reply:
I'm not well versed into Pouch so I just didn't know about this method.

I find it much better than mine!